workflow/util/util.go

```diff
@@ -907,3 +907,29 @@ func GetNodeType(tmpl *wfv1.Template) wfv1.NodeType {
 	}
 	return ""
 }
+
+// MergeWorkflows will do strategic merge the workflows
+func MergeWorkflows(originalWf, patchWf wfv1.Workflow) (*wfv1.Workflow, error) {
```

Comment:
this doesn't actually merge workflows, it merges their specs?

Reply:
if we want to re-use in `setWorkflowDefaults`, which we should because that is the behaviour we want, then we should merge the whole workflow

Reply:
that would add metadata specifically

Reply:
please use this function is `setWorkflowDefaults`

Reply:
> this doesn't actually merge workflows, it merges their specs?

This is generic util function that will merge two workflows. The caller can set elements that they want to merge.  I am changing the setWorkflowDefaults to call this function to merge it.

Reply:
I think it should also merge labels and annotations.